server/src/main/java/org/opensearch/identity/IdentityService.java

```diff
@@ -57,4 +59,8 @@ public Subject getSubject() {
     public TokenManager getTokenManager() {
         return identityPlugin.getTokenManager();
     }
+
+    public void initializeIdentityAwarePlugins(final List<IdentityAwarePlugin> identityAwarePlugins, ThreadPool threadPool) {
```

Comment:
On this subject - it is not needed at all (see please https://github.com/opensearch-project/OpenSearch/pull/14630#discussion_r1720064138)

Reply:
It is needed. I'm running into some issues removing it. Hoping to get a change out shortly.

Reply:
FYI The threadPool is needed because the implementation of NoopPluginSubject will stash the threadcontext similar to plugin usage of threadContext.stashContext today. That way this can be a replacement whether the security plugin is installed or not.

While it may be find to forgo stashing the threadcontext if security is not installed, that introduces a change that is untested in the plugins and I think its better to keep the existing behavior or threadcontext stashing which is thoroughly tested.

The last commit touches a lot of files because the `NoopIdentityPlugin` is not actually a plugin. Its not registered through PluginsService and `createComponents` is never called.

Let me know what you think of this last change, but at some point it probably makes sense to review all instances of where IdentityService is used in tests and create a TestIdentityService that only needs to be modified in a single place.